velox/type/TimestampConversion.cpp

```diff
@@ -447,13 +459,27 @@ bool tryParseTimestampString(
   }
 
   // Try to parse a time field.
-  if (buf[pos] == ' ' || buf[pos] == 'T') {
-    pos++;
-  }
+  switch (parseMode) {
```

Comment:
could move to a helper function, for legibility `parseTimeSeparator(bug, pos, parseMode)`